framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -315,6 +392,61 @@ public boolean isSubtype(AnnotationMirror rhs, AnnotationMirror lhs) {
                     }
                 }
                 return same;
+            } else if (AnnotationUtils.areSameByClass(lhs, DoubleVal.class)
+                    && AnnotationUtils.areSameByClass(rhs, IntRange.class)) {
+                Range rhsRange = getIntRange(rhs);
+                List<Double> lhsValues =
+                        AnnotationUtils.getElementValueArray(lhs, "value", Double.class, true);
+                if (rhsRange.isWiderThan(lhsValues.size())) {
+                    // cannot be subtype if the number of possible values of
```

Comment:
This seems to be always the case if we consider the replacement between IntRange and IntVal. Should we remove this branch?